### PR TITLE
Enable bin/chaperone-server to handle different modes besides HCC

### PR DESCRIPTION
**Pull request by timotree3**
### Changes
- `chaperone-server` now expects a `mode` config parameter that is either `"Chaperone.HCC"` (default), `"Chaperone.DEVELOP"`, or `"Chaperone.Product"`.
- `chaperone-server` now expects a `connection.path` config parameter that defaults to `"/"`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -690,7 +689,7 @@ class Chaperone {
             throw new HoloError("Failed to sign-up");
           }
 
-          if (this.skip_assign_host !== true) {
+          if (this.mode === Chaperone.PRODUCT) {
```
Comment (peeech) [inline src/index.ts:692]: Yeah it looks like `skip_assign_host` is obsolete. In fact host assigning matters only in PRODUCTION mode.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- `chaperone-server` now expects a `mode` config parameter that is either `"Chaperone.HCC"` (default), `"Chaperone.DEVELOP"`, or `"Chaperone.Product"`.
```
Comment (timotree3) [inline CHANGELOG.md:9]:
```suggestion
- `chaperone-server` now expects a `mode` config parameter that is either `"Chaperone.HCC"` (default), `"Chaperone.DEVELOP"`, or `"Chaperone.PRODUCT"`.
```

File: README.md [hunk 3]
```diff
@@ -63,28 +63,29 @@ When configuring Conductor, Chaperone is assuming these conventions:
...
 {
-    "app_id": "app_version",
+    "mode": "Chaperone.HCC",
+    "app_id": "uhCkkmrkoAHPVf_eufG7eC5fm6QKrW5pPMoktvG5LOC0SnJ4vV1Uv",
     "log_level": true,
     "connection": {
-        "secure": false,
-        "port": 42211
-    },
-    "web_user_legend": {
```
Comment (peeech) [inline README.md:72]: I am not sure if removing `web_user_legend` is a good idea. It gets translated into `agent_overrides` that is used heavily in `Chaperone.HCC` and manual tests. @zo-el am I right here?

Comment (timotree3) [reply README.md:72]: I didn't remove it. Just from the example in the readme because I copied that example from `tests/chaperone.json`

Comment (peeech) [reply README.md:72]: True. What I mean removing it from example is not helpful because of this part of usability becoming obscure.

Comment (peeech) [reply README.md:72]: Ah, not true. It's below in description. Sorry!

Comment (timotree3) [reply README.md:72]: No worries. I think you have a point and I would be happy to add it back to the example

Comment (timotree3) [reply README.md:72]: Done

**Review by peeech — APPROVED** [review]
looks great

**Pull request merged**